lib/utf8utils.c

```diff
@@ -24,6 +24,8 @@
 #include "utf8utils.h"
 #include "str-utils.h"
 
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
```

Comment:
Hi @Kokan ,

I didn't find a way to resolve this warning for g_string_append_printf API call at line number:74.
It is because we are passing a format string from a function argument. So, I can not replace that with some constant.